### PR TITLE
feature: add http proxy support

### PR DESCRIPTION
**Pull request by wangjiyang**
Previous implemention didn't support http proxy, which may cause problems when blocked by firewalls. Add http proxy support by refering http_proxy environment variable.

Also fixed a tiny logic error.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -23,13 +23,19 @@ var (
 	client, _    = tls_client.NewHttpClient(tls_client.NewNoopLogger(), options...)
 	access_token = os.Getenv("ACCESS_TOKEN")
 	puid         = os.Getenv("PUID")
+	http_proxy        = os.Getenv("http_proxy")
 )
 
 func main() {
-	if access_token == "" && puid == "" {
+	if access_token == "" || puid == "" {
```
Comment (acheong08) [inline main.go:30]: It works with either of them. You don't need to set both

**Review by acheong08 — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (wangjiyang) [reply main.go:30]: ok. I'll revert this part. Is it reasonable for other parts?

Comment (acheong08) [reply main.go:30]: yes

Comment (wangjiyang) [reply main.go]: Done. Please check my following commit. Thanks.

**Review by acheong08 — APPROVED** [review]

**Pull request merged**